llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -10626,6 +10649,9 @@ SDValue DAGCombiner::visitSRA(SDNode *N) {
   if (SDValue NarrowLoad = reduceLoadWidth(N))
     return NarrowLoad;
 
+  if (SDValue AVG = combineAVG(N))
```

Comment:
We have a "foldAddToAvg" already, perhaps we could name this foldAvgFromShift or combineAvgFromFlags or something like it?

Reply:
foldShiftToAvg then.